packages/help-center/src/contexts/FeatureFlagContext.tsx

```diff
@@ -0,0 +1,20 @@
+import { useContext } from 'react';
+import { FeatureFlags } from '../types';
+
+const FeatureFlagContext = React.createContext( {
+	loadNextStepsTutorial: false,
+} );
+
+export const FeatureFlagProvider: React.FC< FeatureFlags > = function ( { children } ) {
+	// The helpCenterFeatureFlags value is added as a global through the help-center-script
+	return (
+		<FeatureFlagContext.Provider value={ helpCenterFeatureFlags }>
```

Comment:
This is non-blocking and more for curious discussion.

Since `helpCenterFeatureFlags` is a global, what is the point of setting up a context provider and checking the result of the `useFeatureFlags` hook.. when we could just check the `helpCenterFeatureFlags` global instead anyways?  Honestly, I really like being able to use `useFeatureFlags` more than a global... it just feels a lot more clean. Curious about your thoughts though?

Reply:
> Since helpCenterFeatureFlags is a global, what is the point of setting up a context provider and checking the result of the useFeatureFlags hook

Good question! The context + provider allows us to consolidates references to the global to a single spot. We could've done this in many ways, but since feature flags are typically used at different levels of the component tree, context seemed like an idiomatic way to make this information available without all the prop drilling.

Since feature flags will probably be reused throughout the codebase at some point, the useFeatureFlags hook is partly for dev ergonomics and partly so that we can add error handling ( which I haven't done yet )

```
function useFeatureFlags() {
    const context = React.useContext(FeatureFlagContext)
   
    // Trying to call hook outside of component that renders FeatureFlagProvider
    if (context === undefined) {
      throw new Error('useCount must be used within a CountProvider')
    }
    return context
}
```

I learned about it here https://kentcdodds.com/blog/how-to-use-react-context-effectively